modules/govuk_elasticsearch/manifests/housekeeping.pp

```diff
@@ -0,0 +1,64 @@
+# == Class: govuk_elasticsearch::housekeeping
+#
+#  Delete old snapshots from elasticsearch
+#
+#
+# === Parameters:
+#
+# [*es_repos*]
+#   The local elasticsearch repositories where snapshots are stored
+#
+# [*es_snapshot_limit*]
+#   The number of snapshots we want to keep
+#
+# [*user*]
+#   User that invokes backup
+#
+# [*enabled*]
+#   Determines whether this task will run
+#
+# === Variables
+#
+# [*$repositories*]
+#   A string array of Elasticsearch repositories
+#
+class govuk_elasticsearch::housekeeping(
+  $es_repos,
+  $es_snapshot_limit = 5,
+  $user = 'govuk-backup',
+  $enabled = false
+) {
+
+  if $es_repos == undef {
+    fail('No Elasticsearch repositories were set')
+  } else {
+    # Validates a maximum and minimum string length
+    validate_slength($es_repos, 120, 3)
+    $repositories = join(regsubst($es_repos,'(.*)','"\1"'), ' ')
+  }
+
+  if $enabled {
```

Comment:
Can we move this gating to the calling class please and remove this conditional?